tests/integration/test_osm_mysql.py

```diff
@@ -0,0 +1,161 @@
+            status="active",
+            raise_on_blocked=True,
+            timeout=1000,
+        )
+
+        await ops_test.model.relate("osm-keystone:db", f"{APP_NAME}:osm-mysql")
+        await ops_test.model.block_until(lambda: is_relation_joined(ops_test, "db", "osm-mysql"))
+
+        # osm-keystone is initially in blocked status
+        await ops_test.model.wait_for_idle(
+            apps=[APP_NAME, "osm-keystone", "osm-kafka", "osm-zookeeper", "osm-mongodb"],
+            status="active",
+            raise_on_blocked=False,
+            timeout=1000,
+        )
+
+        await ops_test.model.relate("osm-pol:mongodb", "osm-mongodb:mongo")
+        await ops_test.model.block_until(lambda: is_relation_joined(ops_test, "mongodb", "mongo"))
+
+        await ops_test.model.relate("osm-pol:kafka", "osm-kafka:kafka")
+        await ops_test.model.block_until(lambda: is_relation_joined(ops_test, "kafka", "kafka"))
+
+        await ops_test.model.relate("osm-pol:mysql", f"{APP_NAME}:osm-mysql")
+        await ops_test.model.block_until(
+            lambda: is_relation_joined(ops_test, "mysql", "osm-mysql")
+        )
+
+        show_databases_sql = [
+            "SHOW DATABASES",
+        ]
+        get_count_pol_tables = [
+            "SELECT count(*) FROM information_schema.tables WHERE table_schema = 'pol'",
+        ]
+
+        db_unit = ops_test.model.applications[APP_NAME].units[0]
+        server_config_credentials = await get_server_config_credentials(db_unit)
+
+        # Retry until osm-pol runs migrations since it is not possible to wait_for_idle
+        # as osm-pol throws intermittent pod errors (due to being a podspec charm)
+        try:
+            async for attempt in AsyncRetrying(stop=stop_after_attempt(10), wait=wait_fixed(30)):
+                with attempt:
+                    for unit in ops_test.model.applications[APP_NAME].units:
+                        unit_address = await get_unit_address(ops_test, unit.name)
+
+                        # test that the `keystone` and `pol` databases exist
+                        output = await execute_queries_on_unit(
+                            unit_address,
+                            server_config_credentials["username"],
+                            server_config_credentials["password"],
+                            show_databases_sql,
+                        )
+                        assert "keystone" in output
+                        assert "pol" in output
+
+                        # test that osm-pol successfully creates tables
+                        output = await execute_queries_on_unit(
+                            unit_address,
+                            server_config_credentials["username"],
+                            server_config_credentials["password"],
+                            get_count_pol_tables,
+                        )
+                        assert output[0] > 0
+        except RetryError:
+            assert False
```

Comment:
might be worth checking that it has "users with escalated privileges as it has an application (osm-pol) that creates a user and grants it privileges." since this is what makes this relation unique.

Reply:
After double-checking the application (osm-pol), the existence of the `pol` database and tables within it implies that the escalated user has been created correctly (as osm-pol requires privileges to create a user to be able to run the migrations). Additionally, it's hard to test for the existence of a user created by osm-pol as its naming scheme is not straight-forward